### ISSUE TITLE
[redshift] Single node cluster creation fails with cloudformation error

### ISSUE THREAD

**Issue by dirkgomez** [aws/aws-cdk#8887]
When trying to create a single-node cluster in our acceptance environment, creation fails at the cloudformation level:
```
Number of nodes must be not be supplied for cluster type single-node
```

### Reproduction Steps
Here's the relevant code snippet that creates the erroneous cloudformation: 

```
    new redshift.Cluster(this, 
      `Cluster${this.stackName}`,
      {
        ...
        clusterType: environment == 'prd' ? redshift.ClusterType.MULTI_NODE : redshift.ClusterType.SINGLE_NODE,
       ...

### Error Log
```
Number of nodes must be not be supplied for cluster type single-node
```

The created cloudformation templated indeed has a:      NumberOfNodes: 1

### Environment

  - **CLI Version      : ** 1.49.1
  - **Framework Version: ** 1.49.1
  - **Node.js Version:** 14.4.0
  - **OS               :** Mac OS
  - **Language (Version):** TypeScript

### Other

<!-- e.g. detailed explanation, stacktraces, related issues, suggestions on how to fix, links for us to have context, eg. associated pull-request, stackoverflow, gitter, etc -->




--- 

This is :bug: Bug Report

**Comment by nija-at** [comment]
Looks like a bug coming from https://github.com/aws/aws-cdk/blob/8d4c635b5707e0d7cb34a78aed873796a4df7656/packages/%40aws-cdk/aws-redshift/lib/cluster.ts#L428.

This can be worked around by using [property overrides](https://docs.aws.amazon.com/cdk/latest/guide/cfn_layer.html#cfn_layer_raw) and explicitly unsetting the `NumberOfNodes` property.

**Comment by njlynch** [comment]
Dupe of #9961, which was merged two weeks ago and released in 1.62.0.
